tools/build_metrics.py

```
@@ -0,0 +1,443 @@
+        print(f"{duration:>10.2f}  {pct:>5.1f}  {display}")
+
+    print(f"{''*80}")
+    top_total = sum(d for d, _ in display_entries[:top_n])
+    print(
+        f"{'Top entries:':>10}  {top_total:.1f}s ({top_total/compile_time*100:.1f}% of compile time)"
+    )
+
+    # Show link targets if any are significant
+    if link_entries:
+        print(f"\n{''*80}")
+        print(f"Link targets:")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)